Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -347,6 +349,67 @@ Value ArrayPrototype::index_of(Interpreter& interpreter)
     return Value(-1);
 }
 
+Value ArrayPrototype::reduce(Interpreter& interpreter)
+{
+    auto* this_object = interpreter.this_value().to_object(interpreter);
+    if (!this_object)
+        return {};
+
+    auto initial_length = get_length(interpreter, *this_object);
+    if (interpreter.exception())
+        return {};
+
+    size_t start = 0;
+
+    auto accumulator = js_undefined();
+    if (interpreter.argument_count() > 1) {
+        accumulator = interpreter.argument(1);
+    } else {
+        bool start_found = false;
+        while (!start_found && start < initial_length) {
+            auto value = this_object->get_by_index(start);
+            if (interpreter.exception())
+                return {};
+            start_found = !value.is_empty();
+            if (start_found)
+                accumulator = value;
+            start += 1;
+        }
+        if (!start_found) {
+            interpreter.throw_exception<TypeError>(String::format("Reduce of empty array with no initial value"));
+            return {};
+        }
+    }
+
+    auto* callback_function = callback_from_args(interpreter, "reduce");
+    if (!callback_function)
+        return {};
+
+    auto this_value = js_undefined();
+
+    for (size_t i = start; i < initial_length; ++i) {
+        auto value = this_object->get_by_index(i);
+        if (interpreter.exception())
+            return {};
+
+        if (value.is_empty()) {
+            continue;
+        }
```

Comment:
It's not explicitly mentioned in https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md as far as I can see, but Serenity code usually omits curly braces if the body is a single line - i.e.

```cpp
if (value.is_empty())
    continue;
```